index.d.ts

```diff
@@ -65,27 +65,27 @@ declare module 'gdax' {
         size: string;
         product_id: string;
         side: 'buy' | 'sell';
-        stp: 'dc' | 'co' | 'cn' | 'cb';
         type: 'limit' | 'market' | 'stop';
         created_at: string;
         post_only: boolean;
         fill_fees: string;
         filled_size: string;
-        status: 'rejected' | 'received' | 'open' | 'done' | 'pending';
+        status: 'received' | 'rejected' | 'open' | 'done' | 'pending';
         settled: boolean;
         executed_value: string;
+        time_in_force: 'GTC' | 'GTT' | 'IOC' | 'FOK';
     }
 
     export interface OrderResult extends BaseOrderInfo {
-        time_in_force: 'GTC' | 'GTT' | 'IOC' | 'FOK';
-        status: 'rejected' | 'received' | 'open' | 'done';
+        stp: 'dc' | 'co' | 'cn' | 'cb';
     }
 
     export interface OrderInfo extends BaseOrderInfo {
         status: 'received' | 'open' | 'done' | 'pending';
-        funds: number;
-        specified_funds: number;
-        done_at: string;
+        funds?: number;
```

Comment:
Instead of adding ? to all of these, it would be best to make specific types for each type of order that would require the programmer to only have the required fields. Do something that is done with the WebSocket messages:

https://github.com/coinbase/gdax-node/pull/284/commits/068a11c86af16c773e1c6b5f1e0bb527a9562c49#diff-b52768974e6bc0faccb7d4b75b162c99R258

You would have a union type with the different types of orders, market, stop, limit.